store/tikv/client.go

```diff
@@ -125,7 +130,9 @@ func (c *batchCommandsClient) batchRecvLoop(cfg config.TiKVClient) {
 				zap.Stack("stack"))
 			logutil.Logger(context.Background()).Info("restart batchRecvLoop")
 			go c.batchRecvLoop(cfg)
+			return
 		}
+		c.waitGroup.Done()
```

Comment:
it should be put before the `if` statement.

Reply:
No. Only if the goroutine exits without error we need to decrease the wait group.

Reply:
If the goroutine exits with error, the counter in wait group won't decrease. Will the outer side wait for an infinite time under this condition?